packages/eslint-plugin/docs/rules/array-type.mdx

```diff
@@ -58,6 +59,7 @@ const y: readonly string[] = ['a', 'b'];
 ```ts option='{ "default": "generic" }'
 const x: Array<string> = ['a', 'b'];
 const y: ReadonlyArray<string> = ['a', 'b'];
+const z: Readonly<string[]> = ['a', 'b'];
```

Comment:
re https://github.com/typescript-eslint/typescript-eslint/pull/8752#issuecomment-2093144419
(One of) the CI failure(s) is due to this line being flagged, despite being in the "Correct" tab. (you can try it for yourself in the deploy preview by clicking the "Open in playground" button)
```
 FAIL  tests/docs.test.ts (25.146 s, 827 MB heap size)
  ● Validating rule docs › array-type.mdx › code examples ESLint output
    assert(received)
    Expected value to be equal to:
      true
    Received:
      false
    
    Message:
      Expected not to contain lint errors:
    const x: Array<string> = ['a', 'b'];
    const y: ReadonlyArray<string> = ['a', 'b'];
    const z: Readonly<string[]> = ['a', 'b'];
```
Shoutout to @auvred's for his work in https://github.com/typescript-eslint/typescript-eslint/issues/8382 / https://github.com/typescript-eslint/typescript-eslint/pull/8497/files#diff-62842d0740c234c02c26a4ccd3b3fe920b872cff1479f0703c6ec44f9cfb2f3dR407 that caught this!

Reply:
thank you for your advise! 
<img width="975" alt="스크린샷 2024-05-11 오후 11 22 13" src="https://github.com/typescript-eslint/typescript-eslint/assets/102564722/df72a608-7a18-4cae-8095-09ae4fd1913f">

but there is one problem. i use incorrect example`const z: Readonly<string[]> = ['a', 'b'];
` but lint cannot use this type. Is there no way?

Reply:
Hi, sorry for the slow response.

I'm not really familiar with this change, and, I apologize, but I'm having a bit of a hard time understanding what you mean in the question... 

I'm wondering if the issue is that, now that the snapshot has been set, the snapshot tests are failing due to the examples being changed? If so, you may just need to run `cd eslint-plugin; yarn test schemas -u` (or similar); see https://github.com/typescript-eslint/typescript-eslint/issues/8997.

Let me know if that resolves your issue or if the problem is something else! 🙂

Reply:
BTW - looking closer at the screenshot, i think you might also need to run `yarn lint-fix` to fix the existing violations throughout the repo of the new cases that this rule reports.

Reply:
The problem was that I had to apply the changed lint rule(array-type) to another file. Thank you!